modules/calendar/add.php

```diff
@@ -0,0 +1,548 @@
+                          "event_state" => "",
+                          "event_postal" => "" );
+  $locationspec = serialize($locationspecs);
+
+  // Recurring event specifications
+  $recurrspec = array( "event_repeat_freq" => "1", // every
+                       "event_repeat_freq_type" => "0",  // day
+                       "event_repeat_on_num" => "1",
+                       "event_repeat_on_day" => "0",
+                       "event_repeat_on_freq" => "0",
+                       "exdate" => "" );
+  $args = array();
+    // specify values needed for the INSERT into table - libreehr_postcalendar_events - columns
+    $args['form_category'] = "4";  // Vacation
+    $args['new_multiple_value'] = "";
+    $args['form_provider'] = $provider;  // selected provider
+    $args['form_title'] = "Vacation";
+    $args['form_comments'] = "";
+    $args['event_date'] = $event_date;
+    $args['form_enddate'] = $event_end_date;
+    $args['duration'] = 0;  // no duration in provider events
+    $args['form_repeat'] = "1";  // Repeats box checked
+    $args['recurrspec'] = $recurrspec;
+    $args['starttime'] = $starttime;
+    $args['endtime'] = $endtime;
+    $args['form_allday'] = 0;
+    $args['form_apptstatus'] = "-";  // None
+    $args['form_prefcat'] = "0";  // None
+    $args['locationspec'] = $locationspec;
+
+    $def_facility = sqlFetchArray(sqlStatement( "SELECT u.facility_id
+                                                 FROM users u
+                                                 LEFT JOIN facility f on (u.facility_id = f.id)
+                                                 WHERE u.id = ?", array($_SESSION['authId']) ));
+    // default_user_facility denotes id of default facility of logged in user
+    $default_user_facility = $def_facility['facility_id'];
+    // Taking facility & billing facility as default facility of logged in user
+    $args['facility'] = $default_user_facility;
+    $args['billing_facility'] = $default_user_facility;
+
+  // Insert events - Vacation event starts at some date and repeats till end date,
+  // covering Calendar slots over entire clinic hours under selected provider.
+  InsertEvent($args);
+  refreshCalendar(); // after "Add vacation" process is complete
+}
+
+// 2. Clinic Holiday dates
+if ($_POST['form_action'] == "holiday_submit") {
+  if ($_POST['holiday_action'] == "Delete Holiday") {
+    // user requests to delete a set of holidays
+    $dateAndType = json_decode($_POST['jsonData'], true);  // convert JSON string to an associative array
+    // this associative array contains elements in form of "date" => array("day type", "facility id")
+    foreach ($dateAndType as $date => $value) {
+      // for each set of date and facility, delete holiday event under all providers
+      // no need to iterate through providers in this case
+      sqlStatement("DELETE FROM libreehr_postcalendar_events WHERE pc_eventDate = ? AND pc_facility = ?", array($date, $value[1]));
```

Comment:
You need to add the pc_catid here or else you might delete appointments that are there.